src/mon/OSDMonitor.cc

```diff
@@ -5249,11 +5249,16 @@ bool OSDMonitor::preprocess_command(MonOpRequestRef op)
   boost::scoped_ptr<Formatter> f(Formatter::create(format));
 
   if (prefix == "osd stat") {
-    osdmap.print_summary(f.get(), ds, "", true);
-    if (f)
+    if (f) {
+      f->open_object_section("osdmap");
+      osdmap.print_summary(f.get(), ds, "", true);
+      f->close_section();
       f->flush(rdata);
-    else
+    }
+    else {
```

Comment:
please

1. move `else {` into the previous line.
1. no need to pass `f.get()` to `osdmap.print_summary()` if `f` is empty. please just pass `nullptr` to it.

Reply:
I have changed it, thanks for your good advice.